### DISCUSSION TITLE
Show git branch on prompt

### DISCUSSION THREAD

**Discussion by JuergenGuggemos** [nushell/nushell#6414] [Q&A] [answered]
I would like to set my prompt to the same look like this:

![Bildschirmfoto vom 2022-08-25 20-43-26](https://user-images.githubusercontent.com/86912676/186743912-1637e730-7f4d-48e2-8732-6b62a8b347c8.png)

https://github.com/nushell/nushell/blob/main/images/nushell-autocomplete6.gif

But I don't know how. I installed Nushell on Ubuntu 22.04 and Win10. But it looks the same like:

![Bildschirmfoto vom 2022-08-25 20-42-15](https://user-images.githubusercontent.com/86912676/186743720-0d69edb3-f0d1-431f-8c07-2487b7bf2d94.png)

**Comment by fdncred** [comment, marked as answer]
that prompt jt was using there was a customized version from the nu_scripts repo in prompt folder called oh-my.nu. I use it nearly every day in nushell and windows. This is what it looks like.
![image](https://user-images.githubusercontent.com/343840/186758245-8ee5a738-1898-4567-a8fb-27910b0535c3.png)

**Reply by JuergenGuggemos** [reply]
Thanks! Yes, this is what I'm looking for. But how can I install it? Regarding the notes in `oh-my.nu` I should install it with `engine-q`?
`# NOTE: This is meant to run with engine-q and not nushell yet`. I don't habe Rust installed, so I can't compile this.

**Reply by fdncred** [reply]
Let me fix that. I just removed all the references to engine-q. So, read the notes at the top and at the bottom of the script to understand how to use it. Hopefully they're still right.

**Reply by JuergenGuggemos** [reply]
I don't know what I'm doing wrong. I installed the font `JetBrainsMono Nerd Font` and added the following lines into the `config.nu`.

![grafik](https://user-images.githubusercontent.com/86912676/187066470-d0a5e6ef-1803-485b-b22c-597fec8d8ed2.png)

If I start the nushell I got an Error:
![grafik](https://user-images.githubusercontent.com/86912676/187066392-8965205d-4976-4ce2-a0d5-14883b420d55.png)

**Reply by merelymyself** [reply]
I may be wrong, but I think you're missing the gstat plugin. https://github.com/nushell/nushell/tree/main/crates/nu_plugin_gstat

**Reply by fdncred** [reply]
Yup, that's it. You have to install and register the gstat plugin.

**Reply by JuergenGuggemos** [reply]
I have installed the git stat plugin and the plugin is in the folder ~/.cargo/bin/nu_plugin_gstat. But when I execute the command from Nushell: 
`register ~/.cargo/bin/nu_plugin_gstat --encoding msgpack` i got another error message:

![grafik](https://user-images.githubusercontent.com/86912676/187710635-58c30c30-edaa-4622-91d7-b60b3a20ac45.png)

It seems, that the plugin `msgpack` isn't installed:
https://github.com/nushell/nushell/tree/main/crates/nu-plugin/src/serializers

How can I fix this?

**Reply by fdncred** [reply]
msgpack is a recent addition. it may not be used in the version you're using. I think gstat used `-e json` before msgpack.

**Reply by JuergenGuggemos** [reply]
Hmm, I can't install the plugin. I have tried these commands:

register /home/locke/.cargo/bin/nu_plugin_gstat -e json 
register /home/locke/.cargo/bin/nu_plugin_gstat --encoding json 

![grafik](https://user-images.githubusercontent.com/86912676/187986537-04922418-9675-41e2-8cba-49242d32e147.png)

**Reply by fdncred** [reply]
If you're using the latest main branch the encoding will be msgpack.

Oops, it seems you're ping ponging between versions. It's really either msgpack, json, or capnp. There are no other options. 

It might be helpful to delete the $nu.plugin-path. I wonder if it has multiple entries or something.

**Reply by JuergenGuggemos** [reply]
I have installed various versions. Compiled from GitHub, installed with cargo install nu. But it doesn't work. I think I will wait for a stable release and then try again. Thanks for your help.

**Reply by fdncred** [reply]
I'm sorry you're having these problems. I use gstat in Windows, Linux, and MacOS and the only time I have trouble is when the signature for plugins change or I delete my $nu.plugin-path. I wrote a register-plugins.nu script to make it easier to register plugins if you're on the latest main branch.
